homeassistant/components/airtouch5/climate.py

```diff
@@ -0,0 +1,396 @@
+"""AirTouch 5 component to control AirTouch 5 Climate Devices."""
+import logging
+from typing import Any
+
+from airtouch5py.airtouch5_simple_client import Airtouch5SimpleClient
+from airtouch5py.packets.ac_ability import AcAbility
+from airtouch5py.packets.ac_control import (
+    AcControl,
+    SetAcFanSpeed,
+    SetAcMode,
+    SetpointControl,
+    SetPowerSetting,
+)
+from airtouch5py.packets.ac_status import (
+    AcFanSpeed,
+    AcMode,
+    AcPowerState,
+    AcStatus,
+)
+from airtouch5py.packets.zone_control import (
+    ZoneControlZone,
+    ZoneSettingPower,
+    ZoneSettingValue,
+)
+from airtouch5py.packets.zone_name import ZoneName
+from airtouch5py.packets.zone_status import (
+    ZonePowerState,
+    ZoneStatusZone,
+)
+
+from homeassistant.components.climate import (
+    FAN_AUTO,
+    FAN_DIFFUSE,
+    FAN_FOCUS,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    PRESET_BOOST,
+    PRESET_NONE,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, FAN_INTELLIGENT_AUTO, FAN_TURBO
+from .entity import Airtouch5Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Airtouch 5 Climate entities."""
+    client: Airtouch5SimpleClient = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities: list[ClimateEntity] = []
+
+    # Add each AC (and remember what zones they apply to)
+    zone_to_ac: dict[int, AcAbility] = {}
+    for ac in client.ac:
+        for i in range(ac.start_zone_number, ac.start_zone_number + ac.zone_count):
+            zone_to_ac[i] = ac
+        entities.append(Airtouch5AC(client, ac))
+
+    # Add each zone
+    for zone in client.zones:
+        entities.append(Airtouch5Zone(client, zone, zone_to_ac[zone.zone_number]))
```

Comment:
Please correct me if I'm wrong, but `Airtouch5Zone` controls multiple ac in a specific zone, correct?

HA includes areas. In a service call, targeting an area instead of a single device is possible. Can you please test if an area is enough for your use-case? Are there other advantages compared to areas?

Reply:
Not quite. Airtouch 5 is a ducted AC system.
There is one physical AC unit (represented in HA as `Airtouch5AC`).
The air output of this unit is ducted in to multiple rooms, each of which has a damper (to control flow rate) and a temperature sensor. These are represented as an `Airtouch5Zone`.
A zone reports its current temperature, can be turned on and off individually, and you can set the target temperature for it individually.

In HA it ends up looking like:
![image](https://github.com/home-assistant/core/assets/393086/bd247c5d-6a8d-454a-8895-85cf8ea7e12c)
You can set heat/cool mode on the AC unit, and turn each zone on/off by settings them to "Fan only" or "Off".

This mirrors `airtouch4` and `advantage_air`.